### PR TITLE
Fixed some bugs.

### PR DESCRIPTION
**Pull request by jmatholia**
Mostly just bug fixes related to paths and elbows and a workaround for firefox compatibility.

### CODE REVIEW

File: hs/Css/DrawCss.hs [hunk 1]
```diff
@@ -95,23 +114,45 @@ blueCSS = "#blue"? do
...
     margin (px 5) (px 5) (px 5) (px 5)
-    padding (px 2) (px 20) (px 2) (px 20)
+    padding (px 2) (px 2) (px 2) (px 2)
+    width (pct 40)
     roundCorners
     alignCenter
-    fontSize (px 16)
-    border solid (px 2) "#dcdcdc"
+    fontSize (em 0.75)
+    border solid (px 2) "#000000"
```
Comment (david-yz-liu) [inline hs/Css/DrawCss.hs]: use `black`

File: js/draw/draw.js [hunk 2]
```diff
@@ -123,17 +122,17 @@ function makeNode(e) {
...
+        select(document.getElementById('n' + nodeId));
 
         nodeId += 1;
     } else if (mode === 'path-mode') {
         // make elbow joint, only if the dummy point is outside the starting node
-        if (startNode !== null && (position.x < startNode.getAttribute('x') || 
-                                   position.x > parseFloat(startNode.getAttribute('x')) + nodeWidth) &&
-                                  (position.y < startNode.getAttribute('y') || 
-                                   position.y > parseFloat(startNode.getAttribute('y')) + nodeHeight)) {            
+        if (startNode !== null && ((position.x < parseFloat(startNode.getAttribute('x'))) ||
```
Comment (david-yz-liu) [inline js/draw/draw.js]: use `parseFloat(x, 10)` to specify a radix

File: js/draw/draw.js [hunk 3]
```diff
@@ -297,71 +295,54 @@ be a node or an elbow. At least one of beg and end must be a node.
...
-    var theElbow = null;
     var thePath = null;
+    var node1Edges;
+    var node2Edges 
 
     if (typeB === 'node' && typeE === 'elbow') {
-        theNode = beg;
-        theElbow = end;
+        node1Edges = [{x: beg.x + nodeWidth/2, y: beg.y}, 
+                          {x: beg.x + nodeWidth/2, y: beg.y + nodeHeight},
```
Comment (david-yz-liu) [inline js/draw/draw.js]: indentation is off

**Comment by david-yz-liu** [comment]
@jmatholia Just a couple of small bugs, then I'll merge this in

File: js/draw/draw.js [hunk 4]
```diff
@@ -86,8 +79,14 @@ function getPosition(elem) {
     var yPosition = 0;
       
     while (elem) {
-        xPosition += elem.offsetLeft - elem.scrollLeft + elem.clientLeft;
-        yPosition += elem.offsetTop - elem.scrollTop + elem.clientTop;
+        // || 0 -> for mozilla firefox compatability !!
```
Comment (Ian-Stewart-Binks) [inline js/draw/draw.js]: What causes this issue?

Comment (jmatholia) [inline js/draw/draw.js]: I believe elem.offsetLeft and elem.offsetTop are undefined in firefox so when undefined is used in the calculation, the sum is NaN and all the nodes get an x and y of NaN.

**Pull request merged**